modules/combined/tests/temperature_dependent_hardening/temp_dep_hardening_tm.i

```diff
@@ -0,0 +1,286 @@
+[]
+
+[Kernels]
+  [./TensorMechanics]
+    use_displaced_mesh = true
+  [../]
+
+  [./heat]
+    type = HeatConduction
+    variable = temp
+  [../]
+[]
+
+[AuxKernels]
+
+  [./stress_yy]
+    type = RankTwoAux
+    rank_two_tensor = stress
+    variable = stress_yy
+    index_i = 1
+    index_j = 1
+  [../]
+
+  [./total_strain_yy]
+    type = RankTwoAux
+    rank_two_tensor = total_strain
+    variable = total_strain_yy
+    index_i = 1
+    index_j = 1
+  [../]
+
+  [./plastic_strain_xx]
+    type = RankTwoAux
+    rank_two_tensor = plastic_strain
+    variable = plastic_strain_xx
+    index_i = 0
+    index_j = 0
+  [../]
+
+  [./plastic_strain_yy]
+    type = RankTwoAux
+    rank_two_tensor = plastic_strain
+    variable = plastic_strain_yy
+    index_i = 1
+    index_j = 1
+  [../]
+
+  [./plastic_strain_zz]
+    type = RankTwoAux
+    rank_two_tensor = plastic_strain
+    variable = plastic_strain_zz
+    index_i = 2
+    index_j = 2
+  [../]
+[]
+
+
+[BCs]
+  [./y_pull_function]
+    type = FunctionPresetBC
+    variable = disp_y
+    boundary = 3
+    function = top_pull
+  [../]
+
+  [./x_bot]
+    type = PresetBC
+    variable = disp_x
+    boundary = 4
+    value = 0.0
+  [../]
+
+  [./y_bot]
+    type = PresetBC
+    variable = disp_y
+    boundary = 1
+    value = 0.0
+  [../]
+
+  [./z_bot]
+    type = PresetBC
+    variable = disp_z
+    boundary = 0
+    value = 0.0
+  [../]
+
+  [./temp]
+    type = FunctionPresetBC
+    variable = temp
+    function = temp
+    boundary = 4
+  [../]
+[]
+
+[Postprocessors]
+  [./stress_yy_el]
+    type = ElementalVariableValue
+    variable = stress_yy
+    elementid = 0
+  [../]
+[]
+
+[Materials]
+  [./elasticity_tensor]
+    type = ComputeIsotropicElasticityTensor
+    block = 0
+    youngs_modulus = 2e5
+    poissons_ratio = 0.3
+  [../]
+
+  [./small_strain]
+    type = ComputeIncrementalSmallStrain
+    block = 0
+  [../]
+
+  [./temp_dep_hardening]
+    type = RecomputeRadialReturnTempDepHardening
+    block = 0
+    hardening_functions = 'hf1 hf2'
+    temperatures = '300.0 800.0'
+    relative_tolerance = 1e-25
+    absolute_tolerance = 1e-5
+    temp = temp
+  [../]
+
+  [./radial_return_stress]
+    type = ComputeReturnMappingStress
+    block = 0
+    return_mapping_models = 'temp_dep_hardening'
+  [../]
+
+  [./thermal]
+    type = HeatConductionMaterial
+    block = 0
+    specific_heat = 1
+    thermal_conductivity = 1
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+
+  #Preconditioned JFNK (default)
+  solve_type = 'PJFNK'
+
+  petsc_options = '-snes_ksp_ew'
+  petsc_options_iname = '-ksp_gmres_restart -pc_type -pc_hypre_type -pc_hypre_boomeramg_max_iter'
+  petsc_options_value = '201                hypre    boomeramg      4'
+
+  line_search = 'none'
+
+
+  l_max_its = 100
+  nl_max_its = 100
+  nl_rel_tol = 1e-12
+  nl_abs_tol = 1e-10
+  l_tol = 1e-9
+
+  start_time = 0.0
+  end_time = 6
+  dt = 0.1
+[]
+
+[Outputs]
+  csv = true
+  [./out]
+    type = Exodus
+  [../]
+[]
```

Comment:
Since these tests don't really involve a thermal solution, you can make the temperature field an AuxVariable, and use an AuxKernel (FunctionAux, I think) to prescribe a time-dependent temperature history. Then you can move these tests into solid_mechanics and tensor_mechanics instead of having them in the combined module.